nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/transforms/custom/Reverse.java

```diff
@@ -100,8 +101,8 @@ public String tensorflowName() {
 
     @Override
     public List<SDVariable> doDiff(List<SDVariable> f1) {
-        SDVariable ret = f().reverse(f1.get(0), dimensions);
-        return Arrays.asList(ret);
+
+        return Arrays.asList(new ReverseBp(sameDiff, arg(0), sameDiff.reverse(f1.get(0)), dimensions).outputVariables());
```

Comment:
I've very recently (merged to master in last few days) added a DifferentialFunction.outputs() method that returns a List<SDVariable>
So this can now become:
```
return new ReverseBp(sameDiff, arg(0), sameDiff.reverse(f1.get(0)), dimensions).outputs();
```